pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -54,43 +60,101 @@ func (r *reconciler) ReconcileAPIUrl(ctx context.Context, dynakube *dynatracev1b
 }
 
 func (r *reconciler) ReconcileCodeModuleCommunicationHosts(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error {
+	const conditionComponent = "CodeModule"
+
 	log.Info("reconciling istio components for oneagent-code-modules communication hosts")
 
 	if dynakube == nil {
 		return errors.New("can't reconcile oneagent communication hosts of nil dynakube")
 	}
 
+	if !dynakube.NeedAppInjection() {
+		if isIstioConfigured(dynakube, conditionComponent) {
+			log.Info("appinjection disabled, cleaning up")
+
+			return r.CleanupIstio(ctx, dynakube, conditionComponent, OneAgentComponent)
+		} else {
+			return nil
+		}
+	}
+
 	oneAgentCommunicationHosts := oaconnectioninfo.GetCommunicationHosts(dynakube)
 
 	err := r.reconcileCommunicationHostsForComponent(ctx, oneAgentCommunicationHosts, OneAgentComponent)
 	if err != nil {
+		setServiceEntryFailedConditionForComponent(dynakube.Conditions(), conditionComponent, err)
+
 		return err
 	}
 
+	if len(oneAgentCommunicationHosts) > 0 {
+		setServiceEntryUpdatedConditionForComponent(dynakube.Conditions(), conditionComponent)
+	} else {
+		meta.RemoveStatusCondition(dynakube.Conditions(), getConditionTypeName(conditionComponent))
+	}
+
 	return nil
 }
 
 func (r *reconciler) ReconcileActiveGateCommunicationHosts(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error {
+	const conditionComponent = "ActiveGate"
```

Comment:
I don't get the point why we need this one, if we have `ActiveGateComponent` from pkg/controllers/dynakube/istio/config.go:16

Reply:
If I use `ActiveGateComponent` the condition would then be called `IstioForactivegate` which I don't like. 
Uppercasing the first rune is also no good solution imho- adds to code complexity for only mediocre improvement. Simplest solution imho is to use this const.

Reply:
but why not to more to all constants? so later we can refactor it. BTW we use ActiveGateComponent only in 1 place. So it makes sense to tweak it and then just apply strings.Lower(ActiveGateComponent) in 2 places.

Reply:
so something like it:
```diff
--- pkg/controllers/dynakube/istio/config.go
+++ pkg/controllers/dynakube/istio/config.go
@@ -13,7 +13,7 @@ var (
 const (
        OperatorComponent   = "operator"
        OneAgentComponent   = "oneagent"
-       ActiveGateComponent = "activegate"
+       ActiveGateComponent = "ActiveGate"
        IstioGVRName        = "networking.istio.io"
        IstioGVRVersion     = "v1beta1"
 )
```


```diff
 func (r *reconciler) ReconcileActiveGateCommunicationHosts(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error {
-       const conditionComponent = "ActiveGate"
-
        log.Info("reconciling istio components for activegate communication hosts")

        if dynakube == nil {
@@ -106,16 +105,16 @@ func (r *reconciler) ReconcileActiveGateCommunicationHosts(ctx context.Context,
        }

        if !dynakube.NeedsActiveGate() {
-               if isIstioConfigured(dynakube, conditionComponent) {
+               if isIstioConfigured(dynakube, ActiveGateComponent) {
                        log.Info("ActiveGate disabled, cleaning up")

-                       return r.CleanupIstio(ctx, dynakube, conditionComponent, ActiveGateComponent)
+                       return r.CleanupIstio(ctx, dynakube, ActiveGateComponent, strings.ToLower(ActiveGateComponent))
                } else {
                        return nil
                }
        }

-       if !conditions.IsOutdated(r.timeProvider, dynakube, getConditionTypeName(conditionComponent)) {
+       if !conditions.IsOutdated(r.timeProvider, dynakube, getConditionTypeName(ActiveGateComponent)) {
                log.Info("Condition still within time threshold...skipping further reconciliation")

                return nil
@@ -123,17 +122,17 @@ func (r *reconciler) ReconcileActiveGateCommunicationHosts(ctx context.Context,

        activeGateEndpoints := activegate.GetEndpointsAsCommunicationHosts(dynakube)

-       err := r.reconcileCommunicationHostsForComponent(ctx, activeGateEndpoints, ActiveGateComponent)
+       err := r.reconcileCommunicationHostsForComponent(ctx, activeGateEndpoints, strings.ToLower(ActiveGateComponent))
        if err != nil {
-               setServiceEntryFailedConditionForComponent(dynakube.Conditions(), conditionComponent, err)
+               setServiceEntryFailedConditionForComponent(dynakube.Conditions(), ActiveGateComponent, err)

                return err
```

Reply:
done in d56e6ba3

Reply:
done in a9c0115d1d132628574371076b3627808106ae13